gems/smithy/lib/smithy/generators/schema.rb

```diff
@@ -11,26 +11,47 @@ def initialize(plan)
         super
       end
 
-      # @return [Enumerator<String, String>] The file paths and their contents to generate.
       def generate
-        files = source_files
-        files.each do |file, content|
+        gem_files.each_with_object([]) do |(file, content), files|
+          next if file == "lib/#{@gem_name}/customizations.rb" && should_skip_customizations?
+
           create_file file, content
+          files << file
         end
-        files
+      end
+
+      def source
+        source = source_files.map { |_file, content| content }.join("\n")
+        Object.module_eval(source)
+        Object.const_get(@plan.gem_namespace)
+      rescue Exception => e # rubocop:disable Lint/RescueException
+        puts "Error evaluating source:\n#{source}"
+        raise e
       end
 
       private
 
-      def source_files
+      def gem_files
         Enumerator.new do |e|
           e.yield "#{@gem_name}.gemspec", Views::Client::Gemspec.new(@plan).render
           e.yield '.rubocop.yml', Views::Client::RubocopYml.new(@plan).render
```

Comment:
This isn't necessarily related to this change but why is the schema generator using client views?  Additionally - theres a lot of duplication between schema and client generators.

Reply:
Originally I had separation of schema and client views, but they ended up being the same duplicated code. I think some duplication is ok (e.g. usages of these classes) but I think re-defining all of the views was not too necessary. The client and schema gems are effectively the same except the set of files to generate and what gets required. The generator base is responsible for setting up thor and destination file. Originally base shared more logic but with server generation, it doesn't seem the correct abstraction. I think it's correct for these classes to implement the interface of "gem_files" or "source_files"

Reply:
When I think about the these views nested under client namespace, the organization is a bit weird - given that we are reusing them for schema generator (and probably server generator down the road). It could be worthwhile to make specific files such as `types.rb` and `shapes.rb` generic to all generators but I think at the time we were working on this - we were just focused on putting everything under `smithy-client`

Also these `types.rb` and `shapes.rb` has references to `Smithy::Client::...` which is a bit funky - now that we actually want to share the schema gem with early testers - maybe revisit the `smithy-core` gem?